indra/llcommon/lldefs.h

```diff
@@ -168,80 +169,79 @@ const U32	MAXADDRSTR		= 17;		// 123.567.901.345 = 15 chars + \0 + 1 for good luc
 //   llclampb(a)     // clamps a to [0 .. 255]
 //   				   
 
-template <class LLDATATYPE> 
-inline LLDATATYPE llmax(const LLDATATYPE& d1, const LLDATATYPE& d2)
+template <typename T1, typename T2> 
+inline auto llmax(T1 d1, T2 d2)
 {
 	return (d1 > d2) ? d1 : d2;
 }
```

Comment:
The intention there is that the compiler will choose the same type for C that it would choose for:

    auto C = (A > B)? A : B;

or in fact for:

    std::cout << "The larger one is " << ((A > B)? A : B) << std::endl;

I changed the `llmax()` template to allow different datatypes for `d1` and `d2` because the `size_t` changes made many existing `llmax()` expressions fail.